token-group/interface/src/instruction.rs

```diff
@@ -0,0 +1,308 @@
+                buf.extend_from_slice(pod_bytes_of(data));
+            }
+            Self::UpdateGroupAuthority(data) => {
+                buf.extend_from_slice(UpdateGroupAuthority::SPL_DISCRIMINATOR_SLICE);
+                buf.extend_from_slice(pod_bytes_of(data));
+            }
+            Self::InitializeMember(data) => {
+                buf.extend_from_slice(InitializeMember::SPL_DISCRIMINATOR_SLICE);
+                buf.extend_from_slice(pod_bytes_of(data));
+            }
+        };
+        buf
+    }
+}
+
+/// Creates a `InitializeGroup` instruction
+pub fn initialize_group(
+    program_id: &Pubkey,
+    group: &Pubkey,
+    mint: &Pubkey,
+    mint_authority: &Pubkey,
+    update_authority: Option<Pubkey>,
+    max_size: u32,
+    extra_account_metas: &[AccountMeta],
+) -> Instruction {
+    let update_authority = OptionalNonZeroPubkey::try_from(update_authority)
+        .expect("Failed to deserialize `Option<Pubkey>`");
+    let data = TokenGroupInterfaceInstruction::InitializeGroup(InitializeGroup {
+        update_authority,
+        max_size,
+    })
+    .pack();
+    let mut accounts = vec![
+        AccountMeta::new(*group, false),
+        AccountMeta::new_readonly(*mint, false),
+        AccountMeta::new_readonly(*mint_authority, true),
+    ];
+    accounts.extend_from_slice(extra_account_metas);
+
+    Instruction {
+        program_id: *program_id,
+        accounts,
+        data,
+    }
+}
+
+/// Creates a `UpdateGroupMaxSize` instruction
+pub fn update_group_max_size(
+    program_id: &Pubkey,
+    group: &Pubkey,
+    update_authority: &Pubkey,
+    max_size: u32,
+) -> Instruction {
+    let data =
+        TokenGroupInterfaceInstruction::UpdateGroupMaxSize(UpdateGroupMaxSize { max_size }).pack();
+    Instruction {
+        program_id: *program_id,
+        accounts: vec![
+            AccountMeta::new(*group, false),
+            AccountMeta::new_readonly(*update_authority, true),
+        ],
+        data,
+    }
+}
+
+/// Creates a `UpdateGroupAuthority` instruction
+pub fn update_group_authority(
+    program_id: &Pubkey,
+    group: &Pubkey,
+    current_authority: &Pubkey,
+    new_authority: Option<Pubkey>,
+) -> Instruction {
+    let new_authority = OptionalNonZeroPubkey::try_from(new_authority)
+        .expect("Failed to deserialize `Option<Pubkey>`");
+    let data = TokenGroupInterfaceInstruction::UpdateGroupAuthority(UpdateGroupAuthority {
+        new_authority,
+    })
+    .pack();
+    Instruction {
+        program_id: *program_id,
+        accounts: vec![
+            AccountMeta::new(*group, false),
+            AccountMeta::new_readonly(*current_authority, true),
+        ],
+        data,
+    }
+}
+
+/// Creates a `InitializeMember` instruction
+#[allow(clippy::too_many_arguments)]
+pub fn initialize_member(
+    program_id: &Pubkey,
+    group: &Pubkey,
+    group_update_authority: &Pubkey,
+    member: &Pubkey,
```

Comment:
micro-nit: I like to have the params match the instruction
```suggestion
    member: &Pubkey,
    group: &Pubkey,
    group_update_authority: &Pubkey,
```